src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,360 @@
+		clt := createClient()
+		instance := createInstanceWithoutActiveGateService()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := &MockReconciler{}
+		mockCustompropertiesReconciler.On("Reconcile").Return(errors.New(""))
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.Reconcile()
+		mockCustompropertiesReconciler.AssertCalled(t, "Reconcile")
+		require.Error(t, err)
+	})
+	t.Run(`statefulSetReconciler and customPropertiesReconciler error`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstanceWithoutActiveGateService()
+		mockStatefulSetReconciler := &MockReconciler{}
+		mockStatefulSetReconciler.On("Reconcile").Return(errors.New(""))
+		mockCustompropertiesReconciler := &MockReconciler{}
+		mockCustompropertiesReconciler.On("Reconcile").Return(errors.New(""))
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.Reconcile()
+		require.Error(t, err)
+	})
+	t.Run(`Service gets created`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstance()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.Reconcile()
+		mockStatefulSetReconciler.AssertCalled(t, "Reconcile")
+		mockCustompropertiesReconciler.AssertCalled(t, "Reconcile")
+		require.NoError(t, err)
+
+		service := corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, &service)
+
+		assert.NotNil(t, service)
+		assert.NoError(t, err)
+	})
+	t.Run(`Service doesnt get created when missing capabilities`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstanceWithoutActiveGateService()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.Reconcile()
+		mockStatefulSetReconciler.AssertCalled(t, "Reconcile")
+		mockCustompropertiesReconciler.AssertCalled(t, "Reconcile")
+		require.NoError(t, err)
+
+		service := corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, &service)
+
+		assert.Empty(t, service)
+		assert.Error(t, err)
+	})
+}
+
+func TestCreateOrUpdateService(t *testing.T) {
+	t.Run(`createOrUpdateService works`, func(t *testing.T) {
```

Comment:
this is testing the create part of `createOrUpdateService`, please update the test name here